sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java

```diff
@@ -6,22 +6,46 @@
 package com.microsoft.appcenter.utils;
 
 import android.os.Bundle;
-import android.support.test.InstrumentationRegistry;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
 
 /**
  * Wraps InstrumentationRegistry class to enable mocking in unit tests.
  */
 public class InstrumentationRegistryHelper {
 
+    private static final String[] LOCATIONS = new String[]{"androidx.test.platform.app.InstrumentationRegistry",
+            "androidx.test.InstrumentationRegistry",
+            "android.support.test.InstrumentationRegistry"};
+
     /**
      * Get the instrumentation arguments from the InstrumentationRegistry. Wrapper exists for unit
      * tests.
      *
      * @return the instrumentation arguments.
-     * @throws LinkageError          if the class, method is not found or does not match, typically no test dependencies in release.
-     * @throws IllegalStateException if no argument Bundle has been registered.
+     * @throws IllegalStateException if it cannot call getArguments using any of the provided classes.
      */
-    public static Bundle getArguments() throws LinkageError, IllegalStateException {
-        return InstrumentationRegistry.getArguments();
+    public static Bundle getArguments() throws IllegalStateException {
+        Exception exception = null;
+
+        for (String location : LOCATIONS) {
+            try {
+                Class<?> aClass = getClass(location);
+                Method getArguments = aClass.getMethod("getArguments", (Class[]) null);
+                return (Bundle) getArguments.invoke(null, (Object[]) null);
+            } catch (Exception e) {
+                exception = e;
+            }
+        }
+
+        throw new IllegalStateException(exception);
+    }
+
+    public static Class<?> getClass(String className) throws ClassNotFoundException {
```

Comment:
Please make this method package private (no keyword) and move the test class to same package.